pkg/controller/podautoscaler/metrics/metrics_client.go

```diff
@@ -172,69 +204,91 @@ func (h *HeapsterMetricsClient) getCpuUtilizationForPods(namespace string, selec
 		return 0, time.Time{}, fmt.Errorf("failed to unmarshall heapster response: %v", err)
 	}
 
-	if len(metrics.Items) != len(podNames) {
+	// if we don't get full metrics from *all* running pods, make sure we're
+	// not missing metrics from ready pods (if we only compare to the number
+	// of ready pods, we could incorrectly assume we have metrics for all ready
+	// pods, when in reality we have metrics for a mix of ready and unready pods)
+	if len(metrics.Items) != readyPods.Len()+unreadyPods.Len() {
```

Comment:
I think we only have a problem here if we don't error out on pods which are unknown or not running (currently, we don't enter the if, but we catch it below as an "unexpected pod").  It would be nice to not block on pending pods, so I'll do some fiddling.